dotnet/src/Connectors/Connectors.OpenAI/AzureSdk/ClientCore.cs

```diff
@@ -1221,9 +1221,10 @@ private OpenAIChatMessageContent GetChatMessage(ChatChoice chatChoice, ChatCompl
                     arguments = JsonSerializer.Deserialize<KernelArguments>(functionToolCall.Arguments);
                     if (arguments is not null)
                     {
-                        foreach (var argument in arguments)
+                        var names = arguments.Names.ToArray();
```

Comment:
```suggestion
                        // Iterate over copy of the names to avoid mutating the dictionary while enumerating it
                        var names = arguments.Names.ToArray();
```